frontends/p4/typeChecking/typeChecker.cpp

```diff
@@ -104,6 +95,18 @@ TypeChecking::TypeChecking(ReferenceMap* refMap, TypeMap* typeMap,
 
 //////////////////////////////////////////////////////////////////////////
 
+bool TypeInference::learn(const IR::Node* node, Visitor* caller) {
+    auto *learner = clone();
+    learner->setCalledBy(caller);
+    unsigned previous = ::errorCount();
+    (void)node->apply(*learner);
+    unsigned errCount = ::errorCount();
+    bool result = errCount > previous;
+    if (result)
+        typeError("Error while analyzing %1%", node);
+    return result;
```

Comment:
@mbudiu-vmw Would it be possible to simplify this to something like the following?
```C++
    if (errCount > previous)
        typeError("Error while analyzing %1%", node);
    return true;
```